src/components/HorizontalStep/HorizontalStep.tsx

```diff
@@ -0,0 +1,100 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './HorizontalStep.module.css';
+import Icon from '../Icon';
+import Text from '../Text';
+
+export interface Props {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Indicates which number the step is.
+   */
+  stepNumber?: number;
```

Comment:
Can we make this required now since you're always passing it in on line 73 in `HorizontalStepper.tsx`? And then we could also remove the check on 49 of this file?

Reply:
I initially made this optional in case there was use outside of the `HorizontalStepper` but I agree with making this required and crossing the other use cases when we get to it